centreon/tests/php/Core/Common/Infrastructure/RequestParameters/Normalizer/BoolToEnumNormalizerTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Core\Common\Infrastructure\RequestParameters\Normalizer;
+
+use Core\Common\Infrastructure\RequestParameters\Normalizer\BoolToEnumNormalizer;
+
+// Testing nullability
+
+it(
+    'should success for value NULL (when allowed)',
+    fn() => expect((new BoolToEnumNormalizer(nullable: true))->normalize(null))->toBe(null)
+);
+
+// Expected success values
+
+foreach (
+    [
+        // truthy
+        [true, 'customTRUE'],
+        [1, 'customTRUE'],
+        ['true', 'customTRUE'],
+        ['TRUE', 'customTRUE'],
+        ['customTRUE', 'customTRUE'],
+        // falsy
+        [false, 'customFALSE'],
+        [0, 'customFALSE'],
+        ['false', 'customFALSE'],
+        ['FALSE', 'customFALSE'],
+        ['customFALSE', 'customFALSE'],
+    ] as [$tested, $expected]
+) {
```

Comment:
you could use native pestphp datasets instead of foreach : https://pestphp.com/docs/datasets#inline-datasets

Reply:
Good point 👍🏻 I will try to use that.
Thank you 🙏🏻

Reply:
@kduret FYI, I just updated the tests.